src/index.js

```diff
@@ -155,8 +155,14 @@ class ChartComponent extends React.Component {
 	  // use the key provider to work out which series have been added/removed/changed
 	  const currentDatasetKeys = currentDatasets.map(this.props.datasetKeyProvider);
 	  const nextDatasetKeys = nextDatasets.map(this.props.datasetKeyProvider);
+    const shouldWarn = !currentDatasetKeys.every(d => typeof d !== "undefined") || !nextDatasetKeys.every(d => typeof d !== "undefined");
 	  const newDatasets = nextDatasets.filter(d => currentDatasetKeys.indexOf(this.props.datasetKeyProvider(d)) === -1);
 
+    if (shouldWarn && !this.hasWarned) {
```

Comment:
Seems like this will work unless you have a better method https://stackoverflow.com/a/25922668